agent/agent.go

```diff
@@ -21,14 +23,24 @@ import (
 // Start stars the gops agent on a host process. Once agent started,
 // users can use the advanced gops features.
 //
-// Note: The agent exposes an endpoint via a UNIX socket that can be used by
+// Note: The agent exposes an endpoint via a TCP connection that can be used by
 // any program on the system. Review your security requirements before
 // starting the agent.
 func Start() error {
+	gopsdir, err := configdir()
+	if err != nil {
+		return err
+	}
+
 	// TODO(jbd): Expose these endpoints on HTTP. Then, we can enable
 	// the agent on Windows systems.
-	sock := fmt.Sprintf("/tmp/gops%d.sock", os.Getpid())
-	l, err := net.Listen("unix", sock)
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return err
+	}
+	port := l.Addr().(*net.TCPAddr).Port
+	portfile := fmt.Sprintf("%s/.%d", gopsdir, os.Getpid())
```

Comment:
Filename doesn't have to be hidden.

portfile := fmt.Sprintf("%s/%d", gopsdir, os.Getpid())

is fine